libsolidity/ast/Types.cpp

```diff
@@ -406,44 +578,59 @@ bool IntegerConstantType::isImplicitlyConvertibleTo(Type const& _convertTo) cons
 			return true;
 		return false;
 	}
+	else if (_convertTo.category() == Category::FixedPoint)
+	{
+		if (fixedPointType() && fixedPointType()->isImplicitlyConvertibleTo(_convertTo))
```

Comment:
But it's already checking to see if the number can fit into finite precision, is it not?